include/revng/Pipeline/Invokable.h

```diff
@@ -29,8 +29,24 @@ concept HasName = requires() {
   { T::Name } -> convertible_to<const char *>;
 };
 
-// clang-format off
+template<typename InvokableType, typename... Args>
+constexpr bool
+invokableTypeReturnsErrorImpl(void (InvokableType::*F)(Args...)) {
+  return false;
+}
+
+template<typename InvokableType, typename... Args>
+constexpr bool
+invokableTypeReturnsErrorImpl(llvm::Error (InvokableType::*F)(Args...)) {
+  return true;
+}
 
+template<typename Invokable>
+constexpr bool invokableTypeReturnsError() {
+  return invokableTypeReturnsErrorImpl(&Invokable::run);
+}
+
+// clang-format off
```

Comment:
If I understand this correctly, we need this since we use the same machinery to invoke a pipe (returning void) and an analysis (returning `llvm::Error`).
I see from other changes you made that you're trying to let pipes return `llvm::Error` too.
Let's not do that. Pipes return `void` and never fail. Analysis always return `llvm::Error`.